### PR TITLE
Showing google events

### PR DESCRIPTION
**Pull request by mhib**
@adamzima @pbzymek @s1mplex @quirk0o @lkudyba @jtkpiotr

### CODE REVIEW

File: app/assets/javascripts/calendar/calendar.js [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
+        partial.setAttribute('data-id', event.id);
+        partial.setAttribute('class', 'event');
+        partial.setAttribute('style', 'background-color: ' + event.conference_room.color);
+        // TEMPORARY HACK
+        partial.innerHTML =  ['<div class="event-time">', moment(event.start.date_time).format("HH:mm"), '</div>',
+            '<div class="event-name">', event.summary, '</div>',
+            '<div class="event-user">',
+        '<small>by</small>', event.creator.displayName, '</div>',
+            '<div class="event-location">',
+        '<small>in</small>', event.conference_room.title, '</div>'].join('\n');
```
Comment (adamzima) [inline app/assets/javascripts/calendar/calendar.js]: Please format it somehow. Also, why it's temporary?

File: app/controllers/calendar_controller.rb [hunk 2]
```diff
@@ -8,14 +8,15 @@ class CalendarController < ApplicationController
   before_action :refresh_token
   before_action :load_dates_and_rooms, only: [:index, :google_index]
 
-  def index
+  def old_index
```
Comment (adamzima) [inline app/controllers/calendar_controller.rb]: We don't have template for this one

File: app/controllers/calendar_controller.rb [hunk 3]
```diff
@@ -8,14 +8,15 @@ class CalendarController < ApplicationController
...
   end
 
   # Index for showing events from Google calendar
-  def google_index
+  def index
     @events = EventGrouper.new(
-      GoogleEvent.list_events(session[:credentials], DateTime.now.beginning_of_week, DateTime.now.end_of_week)).call
+      GoogleEvent.list_events(session[:credentials], DateTime.now.beginning_of_week, DateTime.now.end_of_week)
+    ).call
     render :index
```
Comment (adamzima) [inline app/controllers/calendar_controller.rb]: We don't need this anymore.

File: app/models/google_event.rb [hunk 4]
```diff
@@ -84,9 +78,31 @@ def calendar_service(credentials)
...
+
+    def new_time_low(time)
+      if time > time.beginning_of_hour + 30.minutes
+        time.beginning_of_hour + 30.minutes
+      else
+        time.beginning_of_hour
+      end
+    end
+
+    def new_time_high(time)
```
Comment (adamzima) [inline app/models/google_event.rb:95]: What's the purpose of this method?

File: app/assets/javascripts/calendar/calendar.js (hunk 1 above)
Comment (mhib) [inline app/assets/javascripts/calendar/calendar.js]: We are going to use react for rendering

File: app/models/google_event.rb (hunk 4 above)
Comment (mhib) [inline app/models/google_event.rb:95]: This is for rendering purposes.
View requires events to have 30 minutes time granularity.

File: app/models/google_event.rb [hunk 5]
```diff
@@ -26,20 +26,14 @@ def list_events(credentials, starting, ending)
...
-              )
+              event.start.date_time = new_time_low event.start.date_time
+              event.end.date_time = new_time_high event.end.date_time
+              events << event.to_h.merge(conference_room: room)
             end
           end
         end
       end
-      events
+      events.sort_by! { |a| a[:end][:date_time] }
```
Comment (jtkpiotr) [inline app/models/google_event.rb]: I would separate sorting logic from listing google events

File: app/services/event_grouper.rb [hunk 6]
```diff
@@ -2,31 +2,62 @@ class EventGrouper
...
   end
 
   def call
-    build_blocks.map { |block| block.sort_by!(&:start_time) }
+    build_blocks.map { |block| block.sort_by! { |n| n[:start][:date_time] } }
   end
 
   private
 
   def build_blocks
```
Comment (jtkpiotr) [inline app/services/event_grouper.rb:14]: Maybe we can better specify what do 'blocks' mean

File: app/services/event_grouper.rb [hunk 7]
```diff
@@ -2,31 +2,62 @@ class EventGrouper
...
     @blocks = [[]]
-
     events.each do |event|
-      add_new_block if event_not_in_block?(@blocks.last, event)
-      @blocks.last << event
+      if (block = find_block_for_event(event))
```
Comment (jtkpiotr) [inline app/services/event_grouper.rb:17]: seems like a C coding style

File: app/controllers/calendar_controller.rb [hunk 8]
```diff
@@ -8,15 +8,12 @@ class CalendarController < ApplicationController
...
-  end
-
   # Index for showing events from Google calendar
-  def google_index
-    @events = EventGrouper.new(
-      GoogleEvent.list_events(session[:credentials], DateTime.now.beginning_of_week, DateTime.now.end_of_week)).call
-    render :index
+  def index
+    @events_hash = GoogleEvent.list_events(session[:credentials], DateTime.now.beginning_of_week, DateTime.now.end_of_week)
+    @events = @events_hash.map do |_wday, events|
```
Comment (mhib) [inline app/controllers/calendar_controller.rb]: You are wrong, m8.

File: app/services/event_grouper.rb (hunk 6 above)
Comment (mhib) [inline app/services/event_grouper.rb:14]: No.

File: app/models/google_event.rb (hunk 4 above)
Comment (adamzima) [inline app/models/google_event.rb:95]: Can we move this `30.minutes` to some constant?

File: app/services/event_grouper.rb (hunk 7 above)
Comment (mhib) [inline app/services/event_grouper.rb:17]: Stop C-shaming

**Pull request merged**